### PR TITLE
DT-158 permission for file copy  and architecture detected

### PR DESCRIPTION
**Pull request by DanielSotoEstrada**
In order to automatically configure Rancher Desktop, a file should be copied into the computer systems of the users. Sometimes this requires additional permissions. So I added the sudo command.

In addition to this we are detecting the architecture of the computer to provide the brew version that will be executed.

In addition to this, we are checking if Rancher Desktop was installed successfully in order to continue with the settings such as symlinks.

### CODE REVIEW

File: lib/core/rancher/step.sh [hunk 1]
```diff
@@ -80,10 +82,9 @@ function ih::setup::core.rancher::install() {
     # Installation and configuration of Rancher Desktop
     for _ in 1 2 3; do
 
-        # Detect Rosetta
-        if [[ $(sysctl -n sysctl.proc_translated) -eq 1 ]]; then
-            # Rosetta  Active
-            arch -arm64 -c brew reinstall ih-rancher
+        # Check if we have a Mac M1 and the terminal is running  over x86
+        if [[ $(sysctl -n sysctl.proc_translated) -eq 1 ]] && [ $(arch) = "i386" ]; then
+            mbrew reinstall ih-rancher
```
Comment (SteveRuble) [inline lib/core/rancher/step.sh:87]: I don't think we can assume `mbrew` alias is available at this point, we should probably just use the real command.

File: lib/core/rancher/step.sh [hunk 2]
```diff
@@ -94,21 +95,27 @@ function ih::setup::core.rancher::install() {
...
-    DOCKERCOMPOSEBIN=/usr/local/bin/docker-compose
-    if [ ! -f "$DOCKERBIN" ]; then
-
-        echo "In order to continue with Rancher configuration and be able to use this engine, some IDEs require the creation of symlinks for remote Python interpreters"
-        echo "Your password is required for the creation of symlink mentioned above"
-        sudo ln -s $HOME/.rd/bin/docker /usr/local/bin/docker
-        if [ ! -f "$DOCKERCOMPOSEBIN" ]; then
-            sudo ln -s $HOME/.rd/bin/docker-compose /usr/local/bin/docker-compose
+    # Continue with the setting just if the cask was installed successfully
+    if [ $CASKSUCCEEDED -eq 0 ]; then
```
Comment (SteveRuble) [inline lib/core/rancher/step.sh:99]: Maybe we should error out here
```shell
if [ $CASKSUCCEEDED -eq 0 ]; then
  ih::log::warn "Could not install Rancher Desktop"
  return 1
fi
```

File: formula/ih-core.rb [hunk 3]
```diff
@@ -1,5 +1,5 @@
 class IhCore < Formula
-  VERSION="0.1.25"
+  VERSION="0.1.26"
```
Comment (SteveRuble) [inline formula/ih-core.rb:2]: Remember to bump this again, I've taken that version.

**Review by SteveRuble — CHANGES_REQUESTED** [review]

File: lib/core/rancher/step.sh (hunk 2 above)
Comment (SteveRuble) [reply lib/core/rancher/step.sh:93]: Arg, this comment is in the wrong spot, and is wrong. It should have been on line 99, and been like this:

```shell
if [ $CASKSUCCEEDED -ne 0 ]; then
  ih::log::warn "Could not install Rancher Desktop"
  return 1
fi
```

**Review by SteveRuble — APPROVED** [review]

**Review by francisco-hurtado — APPROVED** [review]

**Pull request merged**